vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchController.java

```diff
@@ -0,0 +1,34 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.util.HashSet;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.model.impl.ExternalLinkValue;
+import edu.asu.diging.vspace.core.services.IStaffSearchManager;
+
+@Controller
+public class StaffSearchController {
+
+    @Autowired
+    private IStaffSearchManager staffSearchManager;
+    
+    @RequestMapping(value="/staff/search")
+    @GetMapping
+    public String getAllSearchedElements(Model model, @RequestParam(name = "searchText") String searchString) {
+        HashSet<IVSpaceElement> elementList = staffSearchManager.getAllSearchedElements(searchString);
+        model.addAttribute("searchWord", searchString);
+        model.addAttribute("conIndex", new ExternalLinkValue("0"));
```

Comment:
what is "conIndex" and why is it an object of `ExternalLinkValue`?

Reply:
ExternalLinkValue has two method setvalue and getvalue. In staffSearch.html conIndex is used and depending upon various condition setvalue and getvalue methods are called.

Conditions on which conIndex use getvalue and setvalue are as follows:
1.When there are multiple imageblocks of a slide (that matched the searched keyword) only first slide picture will be shown .
2. When there are multiple slide in a module(that matched the searched keyword) only one imageblock from the first slide of the module will be shown corresponding to Module
3. If the module has no slide with image block then a default picture will be shown.

Reply:
what has this todo with `ExternalLinkValue`?

Reply:
There is nothing to do with ExternalLinkValue. Only the code is utilizing the getvalue and setvalue method of this class.

Reply:
I will remove ExternalLinkValue and the change the object type to StringBuilder type.